library/psa_crypto.c

```diff
@@ -1206,12 +1205,10 @@ psa_status_t psa_destroy_key( mbedtls_svc_key_id_t key )
     }
 #endif /* MBEDTLS_PSA_CRYPTO_SE_C */
 
-#if defined(MBEDTLS_PSA_CRYPTO_SE_C)
 exit:
-#endif /* MBEDTLS_PSA_CRYPTO_SE_C */
     status = psa_wipe_key_slot( slot );
     /* Prioritize CORRUPTION_DETECTED from wiping over a storage error */
-    if( overall_status == PSA_SUCCESS )
+    if( status != PSA_SUCCESS )
```

Comment:
Why did you change that? The idea was to give a previous error code the priority over the `psa_wipe_key_slot` error code.

Reply:
The comment did not agree with the code. If there was both a storage or permission error (noted in `overall_status`) and a corruption error (noted in `status`), the comment said to return `status` but the code returned `overall_status`.

Reply:
ah yes indeed.